### PR TITLE
new: Vendor settings update REST api #1192

### PR DESCRIPTION
**Pull request by alamgircsebd**

### CODE REVIEW

**Comment by shanjidah** [comment]
There is one warning that arises when I send some post requests with value.
warning : Notice</b>:  Undefined property: WeDevs\Dokan\REST\StoreSettingController: :$base in
Applications/MAMP/htdocs/dokan1/wp-content/plugins/dokan/includes/REST/StoreSettingController.php

**Comment by alamgircsebd** [comment]
> **There is one warning that arises when I send some post requests with value.**
> **warning : Notice: Undefined property: WeDevs\Dokan\REST\StoreSettingController: :$base in Applications/MAMP/htdocs/dokan1/wp-content/plugins/dokan/includes/REST/StoreSettingController.php**

Resolved already

File: includes/REST/StoreSettingController.php [hunk 1]
```diff
@@ -41,10 +41,48 @@ public function register_routes() {
...
+        $params   = $request->get_params();
+        $store_id = dokan()->vendor->update( $vendor->get_id(), $params );
+
+        if ( is_wp_error( $store_id ) ) {
+            return new WP_Error( $store_id->get_error_code(), $store_id->get_error_message() );
+        }
+
+        $store = dokan()->vendor->get( $store_id );
+
+        do_action( 'dokan_rest_store_settings_update_store', $store, $request );
```
Comment (xaviranik) [inline includes/REST/StoreSettingController.php:78]: Consider changing the name of the hook name `dokan_rest_store_settings_update_store` to `dokan_rest_store_settings_after_update`

File: includes/REST/StoreSettingController.php [hunk 2]
```diff
@@ -83,10 +121,49 @@ protected function get_vendor() {
...
         return $vendor;
     }
 
+    /**
+     * Prepare links for the request.
+     *
+     * @param \WC_Data $object Object data.
+     * @param \WP_REST_Request $request Request object.
+     *
+     * @return array                   Links for the given post.
```
Comment (xaviranik) [inline includes/REST/StoreSettingController.php:136]: Fix the line gap here

File: includes/REST/StoreSettingController.php [hunk 3]
```diff
@@ -83,10 +121,49 @@ protected function get_vendor() {
...
+     */
+    protected function prepare_links( $object, $request ) {
+        $links = [
+            'self'       => [
```
Comment (xaviranik) [inline includes/REST/StoreSettingController.php:140]: Fix indentation here

File: includes/REST/StoreSettingController.php [hunk 4]
```diff
@@ -83,10 +121,49 @@ protected function get_vendor() {
...
+            'collection' => [
+                'href' => rest_url( sprintf( '/%s/%s', $this->namespace, $this->rest_base ) ),
+            ],
+        ];
+
+        return $links;
+    }
+
+    /**
+     * Prepare a single user output for response
```
Comment (xaviranik) [inline includes/REST/StoreSettingController.php:152]: Maybe consider changing the docblock title to **Prepare a single item output for response**

**Review by xaviranik — CHANGES_REQUESTED** [review]

File: includes/REST/StoreSettingController.php [hunk 5]
```diff
@@ -41,10 +41,48 @@ public function register_routes() {
...
+     * @param \WP_REST_Request $request
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @return WP_Error|\WP_REST_Response
+     */
+    public function update_settings( $request ) {
+        $vendor = $this->get_vendor();
+
+        if ( empty( $vendor->get_id() ) ) {
```
Comment (nurul-umbhiya) [inline includes/REST/StoreSettingController.php:65]: kindly move this file into permission check block, also rewrite entire file accordingly (get_settings() methods also needs this checking)

Comment (alamgircsebd) [reply includes/REST/StoreSettingController.php:65]: Resolved

Comment (nurul-umbhiya) [reply includes/REST/StoreSettingController.php:65]: you can safely remove this checking from here, we don't need this checking anymore

**Review by nurul-umbhiya — APPROVED** [review]

**Pull request merged**